nirc_ehr/resources/queries/study/assignment.js

```diff
@@ -2,61 +2,95 @@ require("ehr/triggers").initScript(this);
-    if (prevAnimalId === row.Id) {
-        row.date = prevDate;
+function onInit(event, helper){
+    if (helper.isETL()) {
+        LABKEY.Query.selectRows({
+            schemaName: 'ehr',
+            queryName: 'project',
+            columns: 'project,name',
+            success: function (results) {
+                if (results.rows.length) {
+                    for (var i = 0; i < results.rows.length; i++) {
+                        let rec = results.rows[i];
+                        projectData[rec.name] = rec.project;
+                    }
+                }
+            },
+            scope: this
+        });
     }
+}
 
-    if (row.changeDateText) {
-        if (row.changeDateText.indexOf("ACQ:") === 0) {
-            row.endDate = new Date(Date.parse(row.changeDateText.split(":")[1].trim()));
+EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.BEFORE_INSERT, 'study', 'assignment', function (helper, scriptErrors, row, oldRow) {
+
+    if (helper.isETL()) {
+        if (prevAnimalId === row.Id) {
+            row.date = prevDate;
         }
-        else {
-            row.endDate = new Date(Date.parse(row.changeDateText));
+        else if (count === 0) {
+            // This handles batch boundary row for full truncate ETL, which is the only ETL setup for this currently.
+            // Gets previous enddate from db for first row
+            getLastAssignment(row.Id);
+            if (batchLatestEnddate) {
+                row.date = batchLatestEnddate;
+            }
         }
-    }
 
-    var projectName = row.projectName.split(' ')[0];
-    var projectId = projectData[projectName];
-    if (!projectId) {
-        var projectNames = Object.keys(projectData);
-        for (var i = 0; i< projectNames.length; i++) {
 
-            var pName = projectNames[i] ;
-            if (pName.toString().indexOf(projectName) === 0) {
-                projectId = projectData[projectNames[i]];
+        var projectName = row.projectName.split(' ')[0];
+        var projectId = projectData[projectName];
+        if (!projectId) {
+            var projectNames = Object.keys(projectData);
+            for (var i = 0; i < projectNames.length; i++) {
+
+                var pName = projectNames[i];
+                if (pName.toString().indexOf(projectName) === 0) {
+                    projectId = projectData[projectNames[i]];
+                }
             }
         }
-    }
-    if (projectId) {
-        row.project = projectId;
-    }
-    else {
-        console.log("project can not be found - ", projectName);
-    }
-    prevAnimalId = row.Id;
-    if (!(!row.endDate || row.endDate === 'undefined')) {
-        prevDate = row.endDate;
+        if (projectId) {
+            row.project = projectId;
+        }
+        else {
+            console.log("project can not be found - ", projectName);
+        }
+        prevAnimalId = row.Id;
+        if (!(!row.endDate || row.endDate === 'undefined')) {
```

Comment:
It's unusual to have 'undefined' as a string - is this correct?

Reply:
Yeah it looks like Oracle or the oracle jdbc driver is returning that. Strange, old driver tho